ui/src/components/view/StatsTab.vue

```diff
@@ -371,6 +386,15 @@ export default {
         return
       }
       this.fetchData()
+    },
+    refreshTime: function () {
+      this.fetchData()
+      if (this.refreshTime === '0') return window.clearInterval(this.refreshIntervalId)
+
+      window.clearInterval(this.refreshIntervalId)
+      this.refreshIntervalId = window.setInterval(() => {
+        this.fetchData()
```

Comment:
the start and end dates should be updated before refreshing the data, otherwise we are refreshing the same data everytime.